api/python/t4/packages.py

```diff
@@ -749,16 +747,19 @@ def set(self, logical_key, entry, meta=None):
         Returns:
             self
         """
+        if not logical_key or logical_key.endswith('/'):
```

Comment:
Ah, so we want to allow directory logical keys for attaching metadata to them, we just don't want them to have corresponding physical keys

Reply:
Hm... So what do you pass as `entry` when creating a directory?

Reply:
I assume None. @ResidentMario  ?

Reply:
@dimaryaz Let's make `entry` entirely optional. If left unspecified, try to `set` with the same `logical_key` and `entry` value (e.g. `.set('foo.csv')` == `.set('foo.csv', 'foo.csv')`).

This is something I've been thinking about that would reduce the amount of unnecessary typing when using `set` on regular old files as well. If we do this then we don't need to pass `None` explicitly when using `set` on a directory either. Kills two birds with one stone.

Reply:
Not sure this solves everything, though. What if the directory is not empty?

In any case, do we currently support creating empty directories? We're supposed to, but I can't find the code for that. Right now, I just want to make sure my change isn't breaking anything - and we can implement `None`, etc. as a follow-up.

Reply:
`set_dir` will let you set an empty directory. I don't have a comment on the rest of the changelist because I haven't read it in detail.

Reply:
Anyways... if something is broken, let me know, and I'll fix it.

Reply:
I think you just need to allow the logical key to be None, don't do anything different than the current state. Only raise an exception if the logical key is a directory.